### DISCUSSION TITLE
esp-idf v5.1 and MP Master vs v1.19 and v1.20.0

### DISCUSSION THREAD

**Discussion by gentijo** [micropython/micropython#12022] [ESP32]
I am working on compiling Micro-python for the ESP32, which I have done successfully in the past. 

I am using esp-idf 5.1

When I compile v1.19,  the build fails because it can't find the esp module mdns
When I compile v1.20 the build fails because it can't find the esp module adc

If I compile the tip of main, it compiles

My build system is container based, with 3 batch files, 
00- inits the container, 
01 runs in the container to build the esp-idf, 
02 runs in the container to build micro python.
Defined here https://github.com/gentijo/esp32-micropython-buildenv

Any idea what I am doing wrong to build from a tag ?

Thanks
-John

**Comment by jimmo** [comment]
You cannot build 1.19 or 1.20 with IDF 5.x. There was a fairly large amount of work done after 1.20 to support IDF 5.0 (and now this is the only supported IDF version)

See ports/esp32/README.md (at the appropriate version tags) for supported IDF versions.

**Comment by gentijo** [comment]
Hi Jim,

Thank you, I did read that page but it says 4.4+, should probably add a note 4.4+ and < 5

```
_Note: If you are building MicroPython for the ESP32-S2, ESP32-C3 or ESP32-S3, please ensure you are using the following required IDF versions:
ESP32-S3 currently requires v4.4 or later.
ESP32-S2 and ESP32-C3 require v4.3.1 or later._

```
With that said, I did try to use the 4.x versions of IDF but now I understand that I was probably trying to get them to compile from tip of master, that's why it worked with 5.1. 

Thank you again for the quick reply
-John

**Reply by hemakumarm72** [reply]
it working only v5.0.2
